libp2p/switch.nim

```diff
@@ -75,32 +82,65 @@ type
       streamHandler*: StreamHandler
       secureManagers*: seq[Secure]
       dialLock: Table[PeerID, AsyncLock]
-      ConnEvents: Table[ConnEventKind, HashSet[ConnEventHandler]]
+      connEvents: Table[ConnEventKind, HashSet[ConnEventHandler]]
```

Comment:
do we want to change the hashset here to a seq instead? a bit more predictable in terms of iteration order

Reply:
I wouldn't mind doing that, but it is implied that a `HashSet` preserves order and avoids duplicates. The reason to use a `HashSet`, is to prevent a handler from being registered more than once, we can do the same with a seq, but it would essentially mean duplicating the `HashSet` functionality.

Reply:
hashset is unordered though and might change iteration order of all items on every add

Reply:
Ah, you're right, `HashSet` is unordered, luckily there is an `OrderedSet` in the sets module, we should use that. My bad.